var/spack/repos/builtin/packages/lz4/package.py

```diff
@@ -44,6 +49,39 @@ def url_for_version(self, version):
         else:
             return "{0}/r{1}.tar.gz".format(url, version.joined)
 
+    def patch(self):
+        # Remove flags not recognized by the NVIDIA compiler
+        if self.spec.satisfies("%nvhpc@:20.11"):
+            filter_file("-fvisibility=hidden", "", "Makefile")
+            filter_file("-fvisibility=hidden", "", "lib/Makefile")
+            filter_file("-pedantic", "", "Makefile")
+
+    @run_after("install")
+    def darwin_fix(self):
+        if sys.platform == "darwin":
+            fix_darwin_install_name(self.prefix.lib)
```

Comment:
Once you add builders, these methods should be on the builders (otherwise they are not taken into consideration). I'll check how easy it is to add an audit for stuff like this.